src/createReducer.js

```diff
@@ -303,7 +303,7 @@ function createReducer<M, L>(structure: Structure<M, L>) {
     ) {
       let result = state
       const initial = getIn(result, `initial.${field}`)
-      if (initial === undefined && payload === '') {
+      if (payload === '') {
```

Comment:
to not make this a breaking change for thoose who use `''` to reset value use bellow suggestion
```suggestion
      if ((initial === undefined && payload === '') || payload === undefined) {
```

this will break only 2 tests, fix them to allow `undefined` value

```
  ● reducer.immutable › reducer.immutable.change › should allow setting an initialized field to ''
  ● reducer.immutable › reducer.immutable.change › should not modify a value if called with undefined
```

Reply:
I think only `reset` should reset to `initialValue`, using `change` should allow any value to be changed

Reply:
https://github.com/erikras/redux-form/blob/c89ee4473bc497f2740bc742df9e30b7263419d7/src/__tests__/helpers/reducer.change.js#L189
remove `values` and tests should work.

Reply:
https://github.com/erikras/redux-form/blob/c89ee4473bc497f2740bc742df9e30b7263419d7/src/__tests__/helpers/reducer.change.js#L189
remove `values` and tests should work